### ISSUE TITLE
Create `stack` module

### ISSUE THREAD

**Issue by AndrewJDawes** [codekaizen-github/portainer-ansible-collection#1]
RE the Portainer API Docs:

[Portainer API Docs for Stack](https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/)

From the README.md:

...some parameters for this new `stack` module to expect - along with expectations for how they will drive interactions with Portainer API:

- `state` (enum)
  - `fetched`
    - LIST
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackList
  - `absent`
    - LIST
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackList
      - Look for stack by unique identifier: name
    - DELETE
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackDelete
  - `started`
    - LIST
      - Look for stack by unique identifier: name
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackList
    - POST
      - If the stack does not exist yet, create it
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateKubernetesGit
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateKubernetesFile
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateKubernetesUrl
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerStandaloneFile
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerStandaloneRepository
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerStandaloneString
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerSwarmFile
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerSwarmRepository
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerSwarmString
    - PUT
      - If the stack already exists, update it
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackUpdate
      - If `git`, may also need to call this method
        - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackUpdateGit
    - START
      - Make sure the stack is started
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackStart
  - `stopped`
    - LIST
      - Look for stack by unique identifier: name
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackList
    - POST
      - If the stack does not exist yet, create it
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateKubernetesGit
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateKubernetesFile
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateKubernetesUrl
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerStandaloneFile
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerStandaloneRepository
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerStandaloneString
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerSwarmFile
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerSwarmRepository
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackCreateDockerSwarmString
    - PUT
      - If the stack already exists, update it
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackUpdate
      - If `git`, may also need to call this method
        - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackUpdateGit
    - STOP
      - Make sure the stack is stopped
      - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackStop
- `associate` (bool)
  - After any other changes driven by the `state` param, re-associate the stack with the endpoint passed as arg
  - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackAssociate
  - Basically, give us permission to reassociate the orphaned stack with the endpoint passed as arg
- `redeploy` (bool)
  - After any other changes driven by the `state` param, redeploy the stack
  - If `webhook`, use this strategy
    - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/WebhookInvoke
  - If `git`, use this strategy
    - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackGitRedeploy
- `get_stack_file` (bool)
  - For each stack data returned, add the stack file to the data as an additional attribute
  - https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/stacks/StackFileInspect

**Comment by AndrewJDawes** [comment]
@logansauers When users define a new `stack` they will need to specify an `endpointId`. We may want to discuss whether client code of this Ansible module should be able to optionally specify the Endpoint's name instead of ID (to help with issues of idempotence). If not, then I imagine that anytime this module is used to create a stack, it will need to be preceded by a call to [list out all Endpoints](https://app.swaggerhub.com/apis/portainer/portainer-ce/2.19.5#/endpoints/EndpointList) and find the correct Endpoint ID by name - so this Endpoint ID can later be passed to the `stack` module we are designing here.

So it would just be a decision to abstract this detail for the client code and implement it ourselves under the hood. One thing we would need to double check is whether Endpoint names must be unique. If so, then we can probably abstract the listing of Endpoints. If not, we may have no choice but to require the client to pass in `endpointId`.
